redaxo/src/core/lib/util/i18n.php

```diff
@@ -258,7 +258,7 @@ private static function getMsg($key, $escape, array $replacements, $locale = nul
             }
         }
 
-        $msg = preg_replace($patterns, $replaceWith, $msg);
+        $msg = preg_replace($patterns, $replaceWith, $msg) ?: '';
```

Comment:
Ähnlich wie Diskussion letztens, bin ich nicht so der Fan davon, wegen Typsicherheit hier stillschweigend aus dem Fehler-false einen leeren String zu machen und damit weiterzuarbeiten.
Wobei zugegeben, das Endergebnis hier das gleiche ist.
Aber der Code drückt für mich jetzt halt aus, als ob das explizit so vorgesehen wäre.

Aber wie man am besten damit umgeht, bin ich mir auch nicht 100% sicher. Ob man wirklich überall die `assert()`-Zeile hinzufügt. Oder ob man sich für häufige Funktionen eigene Wrapper schreibt, die die Fehler abfangen. 

Daher meinetwegen lass es hier so. Musste es nur nochmal anmerken ;)

Reply:
assert() an sich macht eigentlich nur sinn wenn wir während der enwicklung die asserts aktivieren. Die logik in assert() wird zur laufzeit am ende in der regel ja nicht laufen

Reply:
Ja ich weiß. Aber wenn ich auf den Code blicke, sehe ich dann, wie es gedacht ist. Und zumindest beim entwickeln wird es auch geprüft.
Bei deiner Variante sieht es für mich wie gesagt so aus, als wäre es legitim, dass preg_replace nicht klappt, und dann mit einem leeren String gearbeitet wird.

Reply:
In eigenen Projekten arbeite ich inzwischen übrigens mit https://github.com/webmozart/assert.

Reply:
Dann sollten wir vllt sowas nehmen https://github.com/thecodingmachine/safe
Ansonsten pflastern wir die codebase mit if(false) throw Exception cases

Reply:
Noch eine Variante wäre, die Stellen erstmal einfach in die baseline aufzunehmen.

Reply:
Aber auch nicht so richtig cool. Die Baseline soll ja eigentlich eher schrumpfen und nicht wachsen.

Reply:
hab an dieser stelle jetzt ne exception